tests/components/input_datetime/test_init.py

```diff
@@ -360,22 +360,33 @@ def storage_setup(hass, hass_storage):
 async def test_reload(hass, hass_admin_user, hass_read_only_user):
     """Test reload service."""
     count_start = len(hass.states.async_entity_ids())
+    ent_reg = await entity_registry.async_get_registry(hass)
 
     assert await async_setup_component(
         hass,
         DOMAIN,
-        {DOMAIN: {"dt1": {"has_time": False, "has_date": True, "initial": "2019-1-1"}}},
+        {
+            DOMAIN: {
+                "dt1": {"has_time": False, "has_date": True, "initial": "2019-1-1"},
+                "dt3": {CONF_HAS_TIME: True, CONF_HAS_DATE: True},
+            }
+        },
     )
 
-    assert count_start + 1 == len(hass.states.async_entity_ids())
+    assert count_start + 2 == len(hass.states.async_entity_ids())
 
     state_1 = hass.states.get("input_datetime.dt1")
     state_2 = hass.states.get("input_datetime.dt2")
+    state_3 = hass.states.get("input_datetime.dt3")
 
     dt_obj = datetime.datetime(2019, 1, 1, 0, 0)
     assert state_1 is not None
     assert state_2 is None
+    assert state_3 is not None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4842434)